File: python/2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        p=l1
        q=l2
        carry=0
        head=current=ListNode(0)
        while(p or q):
            num1 = p.val if p else 0
            num2 = q.val if q else 0
            sum = carry + num1 + num2
            carry = int(sum / 10)
            sum = sum % 10
            current.next = ListNode(sum)
            current = current.next
            p = p.next if p else None
            q = q.next if q else None
        if carry > 0:
            current.next = ListNode(carry)
        return head.next
            

File: java/2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode head = new ListNode(0);
        int carry = 0;
        ListNode current = head;
        while(l1 != null || l2 != null){
            int x = (l1 != null) ? l1.val : 0;
            int y = (l2 != null) ? l2.val : 0;
            int sum = carry + x + y;
            current.next = new ListNode(sum % 10);
            current = current.next;
            carry = sum / 10;
            if (l1 != null) l1 = l1.next;
            if (l2 != null) l2 = l2.next;
        }
        if (carry != 0)
            current.next = new ListNode(carry);
        return head.next;
    }
}